python/ql/src/semmle/python/Concepts.qll

```diff
@@ -526,3 +527,22 @@ module HTTP {
     }
   }
 }
+
+/*
+ */
+
+module RegexExecution {
+  abstract class Range extends DataFlow::Node {
+    DataFlow::Node getRegexNode() {
+      result instanceof DirectRegex or result instanceof CompiledRegex
+    }
+  }
+}
+
+class RegexExecution extends DataFlow::Node {
+  override RegexExecution::Range range;
+
+  RegexExecution() { this = range }
+
+  DataFlow::Node getRegexNode() { result = range.getRegexNode() }
+}
```

Comment:
@yoff is this any closer to what you meant? Apologies for the constant restructuring, but this is the first time I'm doing something this big coding-wise, very fun and exciting though!

Reply:
That'd be great. I'm looking forward to getting a deeper understanding of how the libraries are structured (I wish my programming knowledge was at that level), so that may leave some time to create the experimental structures, and then join this PR's code to it. With that done I want to apply the same changes made in this PR and the former structures knowledge to the rest of the queries by myself to learn further and lower your workload the most (if possible). Thanks 🙌